src/app/static/src/js/components/runWorkflow/runWorkflowSummary.vue

```diff
@@ -1,48 +1,186 @@
 
-    export default Vue.extend<unknown, Computed, unknown, Props>({
+    interface Methods {
+        paramSize: (report: WorkflowReportWithParams) => number;
+        showDetails: (params: Record<string, string>) => Record<string, string>;
+        getReportDependencies: () => void;
+        missingDependencies: (reportName: string) => string[]
+        reportInfo: (report: WorkflowReportWithParams) => string;
+        reportDependsOn: (reportName: string) => string[];
+        hasDependencies: (reportName: string) => boolean
+    }
+
+    interface Data {
+        dependencies: Dependency | null
+        error: string
+    }
+
+    export default Vue.extend<Data, Methods, Computed, Props>({
         name: "runWorkflowSummary",
+        components: {
+            AlertTriangleIcon,
+            InfoIcon
+        },
+        data() {
+            return {
+                dependencies: null,
+                error: ""
+            }
+        },
         props: {
             workflowMetadata: {
                 required: true,
                 type: Object
             }
         },
         computed: {
-            reportCount() {
-                const num = this.workflowMetadata?.reports?.length;
-                if (num == 1) {
-                    return "1 report"
-                } else {
-                    return `${num} reports`
-                }
+            hasMissingDependencies() {
+                return this.dependencies && !!Object.keys(this.dependencies.missing_dependencies)
+                    .some(reportName => this.dependencies.missing_dependencies[reportName].length > 0);
+            }
+        },
+        methods: {
+            reportInfo(report) {
+                return `'${report.name}' runs ${(Object.keys(report.params).length)} times`
+            },
+            paramSize(report) {
+                return Object.keys(report.params).length;
+            },
+            showDetails(params) {
+                return Object.keys(params).slice(3).reduce((entireParams, key) => {
+                    entireParams[key] = params[key]
+                    return entireParams
+                }, {});
+            },
+            getReportDependencies() {
+                api.post(`/workflows/summary`, {
+                    reports: this.workflowMetadata.reports,
+                    ref: this.workflowMetadata.git_commit
+                })
+                    .then(({data}) => {
+                        this.dependencies = data.data;
+                        this.error = "";
+                    })
+                    .catch((error) => {
+                        this.error = error;
+                    })
+            },
+            missingDependencies(reportName) {
+                return this.dependencies?.missing_dependencies[reportName] || [];
+            },
+            reportDependsOn(reportName) {
+                return this.dependencies?.reports.find(report => report.name === reportName)!.depends_on || [];
```

Comment:
Rather than having to search in dependencies (i.e. the response from `/workflows/summary` here, you could use `reports` from that response as the list you render rather the reports from the workflow metadata, so the dependencies would be available in the object already.

Reply:
This would have been ideal but I think there's some sort of race issue. dependencies can not be called until workflowMetadata is available in summary component before a post to get dependencies can be triggered by mounted, if created is used, then workflowMetadata won't be ready.

Reply:
You must have the workflowMetadata as soon as the component is mounted, since that's when you're calling getReportDependencies for the workflow metadata. You would need to be more defensive in the template, since you wouldn't have the reports from the endpoint until that calls returns- so would need to do some `v-if` rather than assuming those reports exist as soon as the component is mounted.

Reply:
`/workflow/summary` does not return params that a user has entered or imported from file, which makes sense because it does not know the params yet. I think workflowMetadata uses the current user data, but we can use (missing dependencies, depends_on) returned from `workflow/summary` endpoint. 

On a second thought, since we have manipulated params at the point of making request to `/workflow/summary`, I guess we should be able to return the updated params.